### PR TITLE
Test on oldest supported versions during end-to-end CI

### LINKED ISSUE #4341: setup.py says we use nose for testing, but we don't
Someone noticed a few weeks back that there's also a dependency on `nose` in one of the requirements file.

We changed to `pytest` and I'm not aware of any `nose` usage anywhere, so let's take it out.

### PR DESCRIPTION
**Pull request by MarcoGorelli**
`pytest.yml` runs all the tests. It's probably a good idea, for that job, to pin all library versions to the earliest possible supported version (as per `requirements.txt`) to make sure that PyMC3 actually works with those versions

----

while I was here, this closes #4341

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -51,7 +51,7 @@ repos:
     - id: pip-from-conda
       additional_dependencies: [pyyaml]
       entry: python scripts/generate_pip_deps_from_conda.py
-      files: ^conda-envs/
+      files: ^conda-envs/environment-dev-py37\.yml$
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml]: It's probably a good idea to have some CI jobs run with different versions of packages, to check PyMC3 works both with the oldest and with the newest versions of its dependencies. So, requirements-dev.txt now has to be generated from just one yml file

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,7 +42,7 @@ repos:
       name: Check no tests are ignored
       pass_filenames: false
     - id: conda-env-sort
-      additional_dependencies: [pyyaml]
+      additional_dependencies: [ruamel.yaml]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml]: `ruamel.yaml` preserves comments (which `pyyaml` destroys) so switching to that

**Review by twiecki — APPROVED** [review]

File: .pre-commit-config.yaml (hunk 2 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml]: will do this in a separate PR

**Review by twiecki — APPROVED** [review]

**Comment by twiecki** [comment]
Thanks @MarcoGorelli!

**Pull request merged**